### PR TITLE
flake: update and add a devShell

### PR DESCRIPTION
**Pull request by ncfavier**
Updates the flake lock to a more recent version of nixpkgs with GHC 9.0.2, and adds a `devShell` to provide some dependencies for building and testing.

### CODE REVIEW

**Comment by andreasabel** [comment]
AFAIK, the `nix` flake is not covered by CI (except for a whitespace check).

@shlevy : Would you be able to review this PR?

**Comment by ncfavier** [comment]
To explain a bit:

The `devShell` attribute is used when you run `nix develop` in the repository, to provide a shell with the required dependencies installed (most notably GHC and cabal). Without a `devShell` attribute, the dependencies are extracted from the default package (in this case Agda). But this is not enough, because for some reason it doesn't include system dependencies like pkg-config, zlib and icu, so I added those explicitly, as well as the fix-whitespace tool for running tests.

The only thing I'm not sure of is whether *more* dependencies could be needed (and I could not realise because they're installed system-wide for me), but those can always be added later.

**Comment by ncfavier** [comment]
It builds now.

File: src/data/emacs-mode/agda2-mode.el [hunk 1]
```diff
@@ -1306,7 +1306,8 @@ top-level scope."
 
 (agda2-maybe-normalised
  agda2-elaborate-give
- "Elaborate check the given expression against the hole's type and fill in hole with the elaborated term"
+ "Elaborate check the given expression against the hole's type and fill in the
+ hole with the elaborated term"
```
Comment (andreasabel) [inline src/data/emacs-mode/agda2-mode.el:1310]: What is the reason for this change?  Is this connected to the nix-build somehow?

Comment (ncfavier) [reply src/data/emacs-mode/agda2-mode.el:1310]: Because of the nixpkgs update, we are now affected by https://github.com/emacs-mirror/emacs/commit/0ebea8ffbfb7b9b1bd92f30011df0875b54eb663 (and warnings are errors)

Comment (andreasabel) [reply src/data/emacs-mode/agda2-mode.el:1310]: Thanks for the clarification!

**Review by andreasabel — APPROVED** [review]

**Review by shlevy — APPROVED** [review]
Thanks!

**Comment by shlevy** [comment]
@ncfavier FYI you can test with `nix develop --ignore-environment` if you want to see if you've gotten all the dependencies.

**Comment by andreasabel** [comment]
@mergifyio rebase
